### PR TITLE
Fix for Bug #490

### PR DESCRIPTION
**Pull request by HuanhuanSunMSFT**
This is partial fix for bug #490. For most prompt for choices, this fix can handle successfully. However, if the prompt choices include one that enable users to enter nested prompt such as the choice called "Suspend", then this fix won't handle it and just throw NotImplementedException. 
It will cost a little bit to implement nested prompt.
@jinglou- @AndreSayreMSFT @MrayerMSFT

### CODE REVIEW

File: PowerShellTools/DebugEngine/HostUi.cs [hunk 1]
```diff
@@ -157,6 +146,9 @@ private string GetPrompt()
     /// </summary>
     public class HostUi
     {
+        [DllImport("user32.dll")]
+        static extern bool SetForegroundWindow(IntPtr hWnd);
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/HostUi.cs]: We start to use this in a few places, including `SetForegroundWindow(Process.GetCurrentProcess().MainWindowHandle);` which is duplicated in a few places.

Did we want to pull these uses and calls out into something more convenient?

File: PowerShellTools/DebugEngine/HostUi.cs [hunk 2]
```diff
@@ -264,6 +256,58 @@ public string ReadLine(string message, string name)
             return input;
         }
 
+        public int ReadChoice(string caption, string message, IList<ChoiceItem> choices, int defaultChoice)
+        {
+            if (String.IsNullOrEmpty(caption))
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/HostUi.cs]: nit: string vs String

File: PowerShellTools/DebugEngine/HostUi.cs [hunk 3]
```diff
@@ -264,6 +256,58 @@ public string ReadLine(string message, string name)
...
+            {
+                message = String.Empty;
+            }
+
+            if (choices == null)
+            {
+                throw new ArgumentNullException("choices");
+            }
+
+            if (choices.Count == 0)
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/HostUi.cs]: nit: !choices.Any()

File: PowerShellTools/DebugEngine/PromptUI/ReadHostPromptForChoicesView.xaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            ShowInTaskbar="False"
+            Title="{Binding Path=Caption}"
+            TextBlock.TextAlignment="Center"
+            WindowStartupLocation="CenterOwner"
+            SizeToContent="WidthAndHeight"
+            d:DataContext="{x:Static local:ReadHostPromptForChoicesViewModel.DesignerViewModel}">
+    <shell:VsShellDialogWindow.Resources>
+        <Style TargetType="Button" BasedOn="{StaticResource {x:Type Button}}">
+            <Setter Property="Height" Value="23" />
+            <Setter Property="Width" Value="75" />
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/PromptUI/ReadHostPromptForChoicesView.xaml]: Confirm: Buttons resize/display properly with long characters/names?

File: PowerShellTools/DebugEngine/ResourceStrings.resx [hunk 5]
```diff
@@ -158,4 +158,10 @@
     <comment>{0} is the Progress Activity [Overall intent, eg, Removing item xyz]
 {1} is the description of the status [eg, 35 of 50 items removed]</comment>
   </data>
+  <data name="ChoicesCollectionShouldNotBeEmpty" xml:space="preserve">
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/ResourceStrings.resx]: Nit: The PowerShell message is 

```
PS C:\WINDOWS\system32> $host.UI.PromptForChoice("test", "test2", @(), 0)
Exception calling "PromptForChoice" with "4" argument(s): ""choices" should have at least one element."
```

("choices" should have at least one element.")

File: PowerShellTools/DebugEngine/HostUi.cs (hunk 2 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/DebugEngine/HostUi.cs]: The guideline is using string when referring to an object and String to a class.

File: PowerShellTools/DebugEngine/HostUi.cs (hunk 3 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/DebugEngine/HostUi.cs]: Resolved.

File: PowerShellTools/DebugEngine/ResourceStrings.resx (hunk 5 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/DebugEngine/ResourceStrings.resx]: Resolved.

File: PowerShellTools/DebugEngine/HostUi.cs (hunk 2 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/DebugEngine/HostUi.cs]: Resolve this to comply with the codeing style here. https://github.com/dotnet/corefx/blob/master/Documentation/coding-style.md

File: PowerShellTools/DebugEngine/HostUi.cs (hunk 1 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/DebugEngine/HostUi.cs]: Done.

**Comment by ASayre** [comment]
[Other than above, looks good!]

File: PowerShellTools/DebugEngine/PromptUI/ReadHostPromptForChoicesViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+using System.Windows.Input;
+using PowerShellTools.Common.Controls;
+using PowerShellTools.Common.ServiceManagement.DebuggingContract;
+
+namespace PowerShellTools.DebugEngine.PromptUI
+{
+    internal sealed class ReadHostPromptForChoicesViewModel
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/PromptUI/ReadHostPromptForChoicesViewModel.cs:12]: also doc comments as well :P

File: PowerShellTools/DebugEngine/DebugServiceEventsHandlerProxy.cs [hunk 7]
```diff
@@ -129,6 +131,11 @@ public string ReadHostPrompt(string message, string name)
             return Debugger.HostUi.ReadLine(message, name);
         }
 
+        public int ReadHostPromptForChoices(string caption, string message, IList<ChoiceItem> choices, int defaultChoice)
```
Comment (ASayre) [inline PowerShellTools/DebugEngine/DebugServiceEventsHandlerProxy.cs:134]: doc comments public methods

**Comment by ASayre** [comment]
Unit tests failed?

**Pull request merged**